coders/uhdr.c

```diff
@@ -531,6 +531,47 @@ static uhdr_mem_block_t GetExifProfile(Image *image, ExceptionInfo *exception)
   return uhdr_profile;
 }
 
+static void fillRawImageDescriptor(uhdr_raw_image_t *imgDescriptor, const ImageInfo *image_info,
+                                   Image *image, void *planes[3], unsigned int strides[3],
+                                   uhdr_img_fmt_t fmt)
+{
+  const char
+    *option;
+
+  imgDescriptor->fmt = fmt;
+
+  if (image->depth >= 10)
+  {
+    option = GetImageOption(image_info, "uhdr:hdr-color-gamut");
+    imgDescriptor->cg = (option != (const char *)NULL) ? map_cg_to_uhdr_cg(option)
+                                                       : getImageColorGamut(&image->chromaticity);
+
+    option = GetImageOption(image_info, "uhdr:hdr-color-transfer");
+    imgDescriptor->ct =
+        (option != (const char *)NULL) ? map_ct_to_uhdr_ct(option) : UHDR_CT_UNSPECIFIED;
+  }
+  else if (image->depth == 8)
+  {
+    option = GetImageOption(image_info, "uhdr:sdr-color-gamut");
+    imgDescriptor->cg = (option != (const char *)NULL) ? map_cg_to_uhdr_cg(option)
+                                                       : getImageColorGamut(&image->chromaticity);
+
+    imgDescriptor->ct = UHDR_CT_SRGB;
+  }
+
+  imgDescriptor->range = UHDR_CR_UNSPECIFIED;
+  imgDescriptor->w = image->columns;
+  imgDescriptor->h = image->rows;
+
+  imgDescriptor->planes[UHDR_PLANE_Y] = planes[0];
```

Comment:
I think it would probably be more readable to do this outside this method for the `planes` and `strides`?

Reply:
The purpose of this function is to populate all the members of rawImgDescriptor structure at once. Will it be inappropriate to fill some structure members inside and some structure members outside the function?

Please let me know your thoughts. I'll modify the function accordingly.

Reply:
I think you can modify some operations outside and some inside the method.

Reply:
Sure, I have made these changes in the recent commit.